### PR TITLE
Feat/calculate password strength

### PR DESCRIPTION
**Pull request by YaninaAQA**

### CODE REVIEW

**Review by vladimir-tl — CHANGES_REQUESTED** [review]
2nd pipeline missing

File: src/calculate-password-strength.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    } else if (strength === 3) {
+
+        return "Weak";
+    } else if (strength === 4) {
+        return "Moderate";
+    } else if (strength >= 5) {
+        return "Strong";
+    }
+    // Default return statement ensuring a string is always returned
+    return "Undefined Strength";
```
Comment (niamh-d) [inline src/calculate-password-strength.ts:28]: The code will never reach this line as anything score 2 or under will be returned as "Very Weak".

File: test/password-strength.test.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import {calculatePasswordStrength} from "../src/calculate-password-strength";
+
+test('Verify that password with strength 2 characters is week', () => {
```
Comment (niamh-d) [inline test/password-strength.test.ts]: This test description should be updated to refer to length.

**Review by niamh-d — CHANGES_REQUESTED** [review]
Nice work! 👍 But would be nice to have one or two more tests.

**Comment by YaninaAQA** [comment]
> 2nd pipeline missing

fixed

**Comment by YaninaAQA** [comment]
> Nice work! 👍 But would be nice to have one or two more tests.

fixed

File: test/password-strength.test.ts (hunk 2 above)
Comment (YaninaAQA) [reply test/password-strength.test.ts]: > This test description should be updated to refer to length.

fixed

File: src/calculate-password-strength.ts (hunk 1 above)
Comment (YaninaAQA) [reply src/calculate-password-strength.ts:28]: without it I have error

**Comment by Marusia2** [comment]
Beautiful:))) Approved:))))

**Pull request merged**